### ISSUE TITLE
Failed to select a version for `winapi`

### ISSUE THREAD

**Issue by tomaka** [rust-lang/cargo#1815]
Trying to compile commit `0c57e48c48f3b329392775af25f627022a1c5f73` of glutin (https://github.com/tomaka/glutin/tree/0c57e48c48f3b329392775af25f627022a1c5f73) produces the following error:

```
failed to select a version for `winapi` (required by `glutin`):
all possible versions conflict with previously selected versions of `winapi`
  version 0.2.1 in use by winapi v0.2.1
  possible versions to select: 0.2.0
```

I am explicitely stating in [the Cargo.toml](https://github.com/tomaka/glutin/blob/0c57e48c48f3b329392775af25f627022a1c5f73/Cargo.toml) that I want winapi `0.2.0` and not `0.2.1`, but cargo doesn't seem to like it.

In the case where I'd be doing something wrong, the error message is not very helping.

**Comment by tomaka** [comment]
Removing the dependencies on the windows `-sys` crates fixes the problem. These crates all depend on `winapi = "*"`.

**Comment by retep998** [comment]
This should have been fixed by https://github.com/rust-lang/cargo/pull/1804

**Comment by tomaka** [comment]
Oh ok. I'll check with tomorrow's nightlies.

**Comment by alexcrichton** [comment]
Yeah I believe this was fixed by #1804, and hopefully new nightlies will come out soon once we get our mac builders under control...

**Comment by tomaka** [comment]
I confirm this is fixed with `5cf4a89 2015-07-21`
